File: typescript/packages/bench/src/rules/signatureRequiredRule.ts
/** Cardinality fixture: nominal (class, `implements`) implementor of RuleContract. */

import type { RuleContract } from "../contracts/ruleContract";
import type { Dataset } from "../dataset";

export class SignatureRequiredRule implements RuleContract {
    readonly key = "signature-required";

    evaluate(data: Dataset): boolean {
        return data.customers.length >= 1;
    }
}

File: rust/crates/atelier-bench/src/rules/signature_required_rule.rs
//! Rule 16/24 (textual half).

use crate::contracts::rule_contract::RuleContract;
use crate::dataset::Dataset;

pub struct SignatureRequiredRule;

impl RuleContract for SignatureRequiredRule {
    fn evaluate(&self, data: &Dataset) -> bool {
        !data.customers.is_empty()
    }
}
